supabase/migrations/20231024141523_create_function_for_fetching_user_balance.sql

```diff
@@ -0,0 +1,9 @@
+create or replace function compute_user_balance(user_id uuid) returns int as $$
+  select sum(
+    case when transaction_type = 'expense' 
+    then (amount * -1) 
+    else amount end) 
+  from transactions
+    where user_id = user_id
+    group by user_id;
```

Comment:
@PCalape I'm not sure if this `group by` statement is needed, can you try without it?

Reply:
Thanks @arnellebalane, it worked well. Applied the changes thanks!